Automattic/wp-calypso
# Site Settings: Introduce a new Protect card

### Description

This PR introduces a new version of the Protect card, aligned with the requirements in p6TEKc-HQ-p2. The legacy Protect card was not compliant with the latest UI and code requirements, and was in need of a substantial refactor, so it made sense to create a new card instead.

The PR is part of #11676.

### Technical implementation
This PR suggests the following changes and additions:

**State**
* Normalize the `jetpack_protect_global_whitelist` field to a simple string.
* Filter Protect module settings when saving settings.
* Tests for the above

**UI**
* Introduce the new Protect card.
* Add the new Protect card to the Security form.
* Remove the legacy Jetpack Protect card and the related README material.

### Preview

**(Protect module active, current IP not in whitelist):**
![](https://cldup.com/XVLuCzEctC.png)

**(Protect module active, current IP in whitelist):**
![](https://cldup.com/oKJWqc0mTw.png)

**(Protect module deactivated):**
![](https://cldup.com/_nPIFO0jJS.png)

### To test

* Checkout this branch
* Go to `/settings/security/$site` where `$site` is a Jetpack site.
* Enable the Protect module from the module toggle in the new card (if it's not enabled).
* Play with the IP field and the "Add to Whitelist" button and try to break them.
* Verify the "Add to Whitelist" button is disabled when the current IP is already whitelisted.
* Verify the IPs save correctly in the database.
* Verify all tests still pass:

```
npm run test-client client/state/jetpack/settings/test/utils.js
```
